src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java

```diff
@@ -555,14 +549,10 @@ public void writeNumberFormat(ByteBuf buf, NumberFormat numberFormat) throws IOE
 
     public PositionSource readPositionSource(ByteBuf buf) {
         PositionSourceType type = PositionSourceType.from(this.readVarInt(buf));
-        switch (type) {
-            case BLOCK:
-                return new BlockPositionSource(this.readPosition(buf));
-            case ENTITY:
-                return new EntityPositionSource(this.readVarInt(buf), buf.readFloat());
-            default:
-                throw new IllegalStateException("Unknown position source type!");
-        }
+        return switch (type) {
```

Comment:
why was the default case removed here?

Reply:
This code will not compile if not all enum values are implemented.